pkg/controllers/gpu/controller.go

```diff
@@ -180,6 +238,23 @@ func (c *controller) UpdateDaemonSet(rawObj runtime.Object, ds *apps.DaemonSet)
 		updated = true
 	}
 
+	if dp.Spec.InitImage == "" {
+		if ds.Spec.Template.Spec.InitContainers != nil {
+			ds.Spec.Template.Spec.InitContainers = nil
+			ds.Spec.Template.Spec.Volumes = removeVolume(ds.Spec.Template.Spec.Volumes, "nfd-source-hooks")
+			updated = true
+		}
+	} else {
+		ds.Spec.Template.Spec.InitContainers = createInitContainer(dp.Spec.InitImage)
+		appendVolumeIfMissing(&ds.Spec.Template.Spec.Volumes, "nfd-source-hooks")
+		updated = true
+	}
+
+	if ds.Spec.Template.Spec.InitContainers[0].Image != dp.Spec.InitImage {
```

Comment:
`ds.Spec.Template.Spec.InitContainers[0].Image` will throw out-of-range in case of `dp.Spec.InitImage == ""`.